buildSrc/src/main/groovy/com.flipkart.varadhi.java-library-conventions.gradle

```diff
@@ -4,4 +4,29 @@ plugins {
 
     // Apply the java-library plugin for API and implementation separation.
     id 'java-library'
+
+    // Apply the maven-publish to allow deploying to maven artifactories
+    id 'maven-publish'
+}
+
+
+publishing {
+    publications {
+        maven(MavenPublication) {
+            groupId = "${projectGroupId}"
+            artifactId = "${project.name}"
+            version = "${version}"
+            from components.java
```

Comment:
What all artifects/components will get published with this ?

Reply:
authz, commons and entities

Reply:
What I was asking is w.r.to  "components.java" ? What all does it publish for a single library module.

Reply:
not sure about this, I saw it was provided as part of publish syntax
published jar has to be maven compatible so it adds the required metadata to be able to correctly indexed by maven repos